src/main/java/de/hysky/skyblocker/skyblock/museum/MuseumManager.java

```diff
@@ -0,0 +1,301 @@
+	private final ToggleButtonWidget prevPageButton;
+	private final TextFieldWidget searchField;
+	private final List<Donation> allDonations;
+	private final List<Donation> filteredDonations = new ArrayList<>();
+	private final List<String> excludedDonationIds = new ArrayList<>();
+	private final List<DonationButton> donationButtons = Lists.newArrayListWithCapacity(BUTTONS_PER_PAGE);
+	private final ButtonWidget filterButton;
+	private final ButtonWidget sortButton;
+	private DonationButton hoveredDonationButton;
+	private int pageCount = 0;
+
+	public MuseumManager(Screen screen, int x, int y, int backgroundWidth) {
+		super(x, y, screen.width, screen.height, Text.of(""));
+		this.layoutX = getX() + backgroundWidth + 2;
+		this.layoutY = y;
+
+		// Initialize search field
+		this.searchField = new TextFieldWidget(TEXT_RENDERER, layoutX + 25, layoutY + 11, SEARCH_FIELD_WIDTH, SEARCH_FIELD_HEIGHT, Text.of("Search"));
+		this.searchField.setMaxLength(60);
+		this.searchField.setVisible(true);
+		this.searchField.setEditableColor(0xFFFFFF);
+		this.searchField.setText(SEARCH_QUERY);
+		this.searchField.setPlaceholder(Text.literal("Search...").formatted(Formatting.ITALIC).formatted(Formatting.GRAY));
+
+		// Initialize page navigation buttons
+		this.nextPageButton = new ToggleButtonWidget(layoutX + 93, layoutY + 133, 12, 17, false);
+		this.nextPageButton.setTextures(RecipeBookResults.PAGE_FORWARD_TEXTURES);
+		this.prevPageButton = new ToggleButtonWidget(layoutX + 38, layoutY + 133, 12, 17, true);
+		this.prevPageButton.setTextures(RecipeBookResults.PAGE_BACKWARD_TEXTURES);
+
+		this.allDonations = MuseumItemCache.getDonations();
+
+		// Create donation buttons for pagination
+		for (int i = 0; i < BUTTONS_PER_PAGE; i++) {
+			DonationButton button = new DonationButton(layoutX + 11 + 31 * (i % 4), layoutY + 34 + 31 * (i / 4));
+			this.donationButtons.add(button);
+		}
+
+		// Initialize sort button
+		this.sortButton = ButtonWidget.builder(Text.literal(""), button -> {
+					ITEM_SORTER.cycleSortMode(filteredDonations);
+					button.setTooltip(ITEM_SORTER.getTooltip());
+					CURRENT_PAGE = 0;
+					updateButtons();
+				})
+				.tooltip(ITEM_SORTER.getTooltip())
+				.position(layoutX + 95, layoutY + 11)
+				.size(BUTTON_SIZE, BUTTON_SIZE)
+				.build();
+
+		// Initialize filter button
+		this.filterButton = ButtonWidget.builder(Text.literal(""), button -> {
+					ITEM_FILTER.cycleFilterMode(allDonations, filteredDonations);
+					ITEM_SORTER.applySort(filteredDonations);
+					button.setTooltip(ITEM_FILTER.getTooltip());
+					CURRENT_PAGE = 0;
+					updateButtons();
+				})
+				.tooltip(ITEM_FILTER.getTooltip())
+				.position(layoutX + 116, layoutY + 11)
+				.size(BUTTON_SIZE, BUTTON_SIZE)
+				.build();
+
+		ITEM_FILTER.applyFilter(allDonations, filteredDonations);
+		ITEM_SORTER.applySort(filteredDonations);
+		updateSearchResults(false);
+
+		((ScreenAccessor) screen).callAddDrawableChild(this);
```

Comment:
Fabric API provides `Screens#getButtons` and adding to the returned list should be equivalent to this